### PR TITLE
Add schema for ansible-navigator

### PR DESCRIPTION
**Pull request by ziegenberg**
This is work-in-progress, but if there's something completely wrong I welcome comments.

Closes: https://github.com/ansible/ansible-navigator/issues/188
Related: https://github.com/ansible/vscode-ansible/issues/328

### CODE REVIEW

**Comment by ziegenberg** [comment]
Why do they use dashes instead of underscores for the ansible-navigator configuration file? :(

**Comment by ziegenberg** [comment]
Ok, I think I got a first version that's ready to be discussed. Tests are green and use the example settings file from https://ansible-navigator.readthedocs.io/en/latest/settings/#the-ansible-navigator-settings-file

**Comment by ssbarnea** [comment]
That is awesome and we will discuss it for sure, it was already on my todo list so that cames as a really nice xmas present.

File: src/ansibleschemas/navigator.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
+# Used to generate JSON Validations chema for ansible-navigator config files
+# https://github.com/ansible-community/ansible-lint/blob/master/.ansible-lint
```
Comment (webknjaz) [inline src/ansibleschemas/navigator.py:2]: Navigator?

**Comment by webknjaz** [comment]
> Why do they use dashes instead of underscores for the ansible-navigator configuration file? :(

It's quite common and kebab-case is nicer in places where snake_case is not required for things to be compatible with Python identifiers.

File: src/ansibleschemas/navigator.py (hunk 1 above)
Comment (ziegenberg) [reply src/ansibleschemas/navigator.py:2]: That's copy&paste error. I'll fix it

Comment (ssbarnea) [reply src/ansibleschemas/navigator.py:2]: Lets wait for @cidrblock on this one, he knows navigator better than anyone else and may be able to spot some schema mistakes.

Comment (ziegenberg) [reply src/ansibleschemas/navigator.py]: Fixed it with 550100a.

**Comment by ziegenberg** [comment]
Rebased onto current `main`.

**Review by ssbarnea — APPROVED** [review]

**Review by ganeshrn — APPROVED** [review]
Thank you!

**Review by priyamsahoo — APPROVED** [review]

**Comment by cidrblock** [comment]
@ziegenberg TY for this btw..... It made my day :)

**Comment by ziegenberg** [comment]
@cidrblock You are welcome :)

**Pull request merged**